---
title: "Group 2: ISS Presentation"
author: 
- Adi (s212940)
- Prince (s220520)
- Razia (s151549)
- Yuan (s212420)
output: ioslides_presentation
widescreen: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE) 
```

```{r Load library, include=FALSE}
library(tidyverse)
library(patchwork)
library(DiagrammeR)
library(kableExtra)
```

## Introduction to dataset 
- Nature article: Microbial composition on the International Space Station
- Article plots them in stacked barplots 
- Two datasets: no. reads per sample, sorted by Taxonomic classification (Domain, Phylum, Genus)
- Sorted into sampling session (A,B,C)

## SD1
- Supplementary data 1 (fig 1): RSV table (ribosomal sequence variants - OTU alternative)
```{r echo=FALSE, message=FALSE, warning=FALSE}
read_csv2(file = "/cloud/project/data/SD1_converted.csv") %>% 
  head() %>% 
  select(`Taxonomic classification`) %>% 
  kbl() %>% 
  kableExtra::kable_minimal()  
```
## SD2
- Supplementary data 2 (fig 4): Taxonomic diversity inferred from metagenomic dataset  
```{r echo=FALSE, message=FALSE, warning=FALSE}
read_csv2(file = "/cloud/project/data/SD2_converted.csv") %>% 
  head() %>% 
#  select(`Taxonomic classification`) %>% 
  kbl() %>% 
  kableExtra::kable_minimal() 
```
## Table 1
- Table 1 (tidy): sampling locations and sampling sessions
```{r echo=FALSE, message=FALSE, warning=FALSE}
read_csv2(file = "/cloud/project/data/table1_tidy.csv") %>% 
#  head() %>% 
  select(-1) %>% 
  slice(1:10) %>% 
  kbl() %>% 
  kableExtra::kable_minimal() 
```

## Data Processing Workflow
```{r Data Processing Flowchart }
DiagrammeR::grViz("digraph {
  graph [layout = dot, rankdir = TB]
  
  node [shape = rectangle]        
  rec1 [label = 'Import: .xlsx -> .csv']
  rec2 [label = 'Tidy: (remove unused Tax IDs)']
  rec3 [label =  'Wrangle: Normalise & rename columns, pivot_longer()']
  rec4 [label = 'Visualise: ggplot2()']
  
  # edge definitions with the node IDs
  rec1 -> rec2 -> rec3 -> rec4
  }",
  height = 500)
```

# Results & Discussion
## Figure 1: Domain
```{r fig.show = "hold", out.width = "90%", fig.align = "center"} 
knitr::include_graphics(c("images/fig1_Domain.png",
                        "images/Article_fig1_Domain.png"))
```

## Figure 1: Phylum
```{r fig.show = "hold", out.width = "90%", fig.align = "center"} 
knitr::include_graphics(c("images/fig1_Phylum.png",
                        "images/Article_fig1_Phylum.png"))
```

## Figure 1: Genus
```{r fig.show = "hold", out.width = "90%", fig.align = "center"} 
knitr::include_graphics(c("images/fig1_Genus.png",
                        "images/Article_fig4_Genus.png"))
```

## Figure 1 discussion
- Phylum: they have an ‘unknown’ section, but we can’t find that in the data
- Genus: They haven’t normalised, they haven’t provided clean room data, they have multiple sections in barograph not shown in legend, their colour choice is bad: unknown and lactobacillus are very difficult to distinguish


## Figure 4: Domain
```{r fig.show = "hold", out.width = "90%", out.height = "1%", fig.align = "center"} 
knitr::include_graphics(c("images/fig4_Domain.png",
                        "images/Article_fig4_Domain.png"))
```

## Figure 4: Phylum
```{r fig.show = "hold", out.width = "90%", fig.align = "center"} 
knitr::include_graphics(c("images/fig1_Phylum.png",
                        "images/Article_fig4_Phylum.png"))
```

## Figure 4: Genus
```{r fig.show = "hold", out.width = "90%", fig.align = "center"} 
knitr::include_graphics(c("images/fig1_Genus.png",
                        "images/Article_fig4_Genus.png"))
```

## Figure 4 Discussion
- *General*: Their use of `sqrt(TSS)` makes no sense
- Method of selection unclear; solution: sorted top 100
- Domain: ‘other sequences’ not visible in out plot, but there is only 1 read 
```{r echo=FALSE, message=FALSE, warning=FALSE}
read_csv2(file = "/cloud/project/data/SD2_clean_TSS.csv") %>% 
  count(domain) %>% 
  kbl() %>%
  kable_styling()
```
- Genus: we filtered out all non-bacteria 


